### PR TITLE
More builtins

### PR DESCRIPTION
**Pull request by ChristianoBraga**
Added all builtins up to ByteString handling, according to v2.1 of Plutus Core specification.
There are 4 tests at tests/simple that are still failing. The ifThenElse builtin seems to require force to be evaluated, according to uplc, but the semantics does not seem to require that. Builtins greaterThanInteger and greaterThanEqualsInteger are not implemented in uplc. Finally, it seems that one may not write a (error) term. It appears to be something that is issued by the runtime only. Again, the semantics does not imply that.

### CODE REVIEW

File: tests/simple/conByteString.uplc.krun.expected [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <generatedTop>
   <k>
-    ( con bytestring #00FF ) ~> .
+    ( con bytestring #00ff ) ~> .
```
Comment (SchmErik) [inline tests/simple/conByteString.uplc.krun.expected:3]: I think you forgot to add the output from uplc for this test. In otherwords, I'm not seeing a `conByteString.uplc.expected` file. I have a fleeting question: does the grammar supported by uplc support capital letters A-F as well as a-f?

File: tests/simple/error.uplc.expected [hunk 2]
```diff
@@ -1,11 +0,0 @@
-<generatedTop>
```
Comment (SchmErik) [inline tests/simple/error.uplc.expected:1]: I know that uplc emits an error on this that's emitted on stderr. @ChristianoBraga and @ehildenb should we capture the output of stderr when we capture expected results?

File: tests/simple/force.uplc [hunk 3]
```diff
@@ -1 +1 @@
-(program 1.0.0 (force x))
```
Comment (SchmErik) [inline tests/simple/force.uplc:1]: again, I think we might want to keep tests that involve free variables and create another test with `lam x (force x)`

File: tests/simple/var.uplc [hunk 4]
```diff
@@ -1 +1 @@
-(program 1.0.0 x)
```
Comment (SchmErik) [inline tests/simple/var.uplc:1]: I think we should keep this test `(program 1.0.0 x)`

**Review by SchmErik — CHANGES_REQUESTED** [review]

File: tests/simple/delay.uplc [hunk 5]
```diff
@@ -1 +1 @@
-(program 1.0.0 (delay x))
```
[the remark this answers is not in the document]
Comment (ChristianoBraga) [reply tests/simple/delay.uplc:1]: I added `delay.uplc` back to `tests/simple` and also a line for it in `tests/failing`.
I also added file `delay-lam.uplc` (using make `update-results`) which should pass.
Lines starting with `#` in tests/failing will not be considered.

File: tests/simple/conByteString.uplc.krun.expected (hunk 1 above)
Comment (ChristianoBraga) [reply tests/simple/conByteString.uplc.krun.expected:3]: Added.
Yes, uplc supports both small and big caps.

File: tests/simple/error.uplc.expected (hunk 2 above)
Comment (ChristianoBraga) [reply tests/simple/error.uplc.expected:1]: Yes, just have to change the semantics to generate what has caused the error. In the `error.uplc` is empty 
```
An error has occurred:  User error:
The provided Plutus code called 'error'.
``` 
but `(program 0.0.0 [[(builtin divideInteger) (con integer 1)] (con integer 0)])` produces 
```
An error has occurred:  User error:
The provided Plutus code called 'error'.
Caused by: (divideInteger 1 0)
```

File: tests/simple/force.uplc (hunk 3 above)
Comment (ChristianoBraga) [reply tests/simple/force.uplc:1]: Added `force-lam.uplc` and included `force.uplc` in the failing list.

File: tests/simple/var.uplc (hunk 4 above)
Comment (ChristianoBraga) [reply tests/simple/var.uplc:1]: Same as force and delay.

**Pull request merged**